src/recovery_status_parser.rs

```diff
@@ -71,23 +71,103 @@ where
         })
 }
 
-fn target_status<I>() -> impl Parser<I, Output = Record>
+fn clients_line<I>(x: &'static str) -> impl Parser<I, Output = u64>
+where
+    I: Stream<Token = char>,
+    I::Error: ParseError<I::Token, I::Range, I::Position>,
+{
+    (
+        attempt(string(x).skip(optional(token(':')))),
+        spaces(),
+        digits(),
+        optional((token('/'), digits())),
+        optional(newline().map(drop).or(eof())),
+    )
+        .map(|(_, _, x, _, _): (_, _, u64, _, _)| x)
+}
+
+#[derive(Debug)]
+enum RecoveryStat {
+    Status(RecoveryStatus),
+    Completed(u64),
+    Connected(u64),
+    Evicted(u64),
+}
+
+fn target_recovery_stats<I>() -> impl Parser<I, Output = Vec<RecoveryStat>>
+where
+    I: Stream<Token = char>,
+    I::Error: ParseError<I::Token, I::Range, I::Position>,
+{
+    many(choice((
+        status_line()
+            .skip(optional(newline()))
+            .map(RecoveryStat::Status)
+            .map(Some),
+        clients_line("completed_clients")
+            .skip(optional(newline()))
+            .map(RecoveryStat::Completed)
+            .map(Some),
+        clients_line("connected_clients")
+            .skip(optional(newline()))
+            .map(RecoveryStat::Connected)
+            .map(Some),
+        clients_line("evicted_clients")
+            .skip(optional(newline()))
+            .map(RecoveryStat::Evicted)
+            .map(Some),
+        // This will ignore line/field we don't care
+        attempt((target(), token(':'), till_newline().skip(newline()))).map(|_| None),
```

Comment:
Which lines does this match?

Is there any case where it can accidentally match something unintentional?

Reply:
It will match any lines that starts with alphanum and `_`/`-` and contains `:`.
I haven't found any new stats line that matches this. The closest is 
```
obdfilter.ai400-OST0000.job_stats=job_stats:
```
But since it contains `.` and `=`, it will not be matched.

All new stats uses `=` so it will never match against those.

Reply:
But we aren't actually matching a target name here, are we?

This parser should be typed explicitly instead of reusing target parser for a different purpose.

If we ever change target to be smarter on what it matches, this will break.

Reply:
I see, let me push a patch to create a new function for this specific instead of re-using the `target` one.